pkg/component/autoscaling/gardenercustommetrics/component.go

```diff
@@ -0,0 +1,232 @@
+}
+
+// NewGardenerCustomMetrics creates a new GardenerCustomMetrics instance tied to a specific server connection.
+//
+// namespace is where the server-side artefacts (e.g. pods) will be deployed (usually the 'garden' namespace).
+// containerImageName points to the binary for the gardener-custom-metrics pods. The exact version to be used, is
+// determined by contextual configuration, e.g. image vector overrides.
+// If enabled is true, this instance strives to bring the component to an installed, working state. If enabled is
+// false, this instance strives to uninstall the component.
+// seedClient represents the connection to the seed API server.
+// secretsManager is used to interact with secrets on the seed.
+func NewGardenerCustomMetrics(
+	namespace string,
+	containerImageName string,
+	kubernetesVersion *semver.Version,
+	seedClient client.Client,
+	secretsManager secretsmanager.Interface) *GardenerCustomMetrics {
+	return &GardenerCustomMetrics{
+		namespaceName:      namespace,
+		containerImageName: containerImageName,
+		kubernetesVersion:  kubernetesVersion,
+		seedClient:         seedClient,
+		secretsManager:     secretsManager,
+
+		testIsolation: gardenerCustomMetricsTestIsolation{
+			CreateForSeed: managedresources.CreateForSeed,
+			DeleteForSeed: managedresources.DeleteForSeed,
+		},
+	}
+}
+
+// Deploy implements [component.Deployer.Deploy]()
+func (gcmx *GardenerCustomMetrics) Deploy(ctx context.Context) error {
+	baseErrorMessage :=
+		fmt.Sprintf(
+			"An error occurred while deploying gardener-custom-metrics component in namespace '%s' of the seed server",
+			gcmx.namespaceName)
+
+	serverCertificateSecret, err := gcmx.deployServerCertificate(ctx)
+	if err != nil {
+		return fmt.Errorf(baseErrorMessage+
+			" - failed to deploy the gardener-custom-metrics server TLS certificate to the seed server. "+
+			"The error message reported by the underlying operation follows: %w",
+			err)
+	}
+
+	kubeObjects, err := kubeobjects.GetKubeObjectsAsYamlBytes(
+		deploymentName, gcmx.namespaceName, gcmx.containerImageName, serverCertificateSecret.Name, gcmx.kubernetesVersion)
+	if err != nil {
+		return fmt.Errorf(baseErrorMessage+
+			" - failed to create the K8s object definitions which describe the individual "+
+			"k8s objects comprising the application deployment arrangement. "+
+			"The error message reported by the underlying operation follows: %w",
+			err)
+	}
+
+	err = gcmx.testIsolation.CreateForSeed(
+		ctx,
+		gcmx.seedClient,
+		gcmx.namespaceName,
+		managedResourceName,
+		false,
+		kubeObjects)
+	if err != nil {
+		return fmt.Errorf(baseErrorMessage+
+			" - failed to deploy the necessary resource config objects as a ManagedResource named '%s' to the server. "+
+			"The error message reported by the underlying operation follows: %w",
+			managedResourceName,
+			err)
+	}
+
+	return nil
+}
+
+// Destroy implements [component.Deployer.Destroy]()
```

Comment:
```suggestion
```

Reply:
This comment has two purposes:
- It documents the intent, to help in refactoring scenarios where conformance to the interface is unintentionally broken
- In IDEs, it provides for a hint which can be conveniently navigated to the documentation of the interface method (parameters, return values, etc.)

Reply:
My motivation for this suggestion was that most of the components don't have doc strings for func that they implement from the component interfaces (Deploy/Destroy). Anyways, if you insist you can keep the doc string as I see that we are not very consistent wrt across the codebase - there are components that have doc strings for these funcs so that's why I won't insist on this comment.

However, I would suggest:
```suggestion
// Destroy implements component.Deployer.Destroy.
```

I am not sure what the `[foo]()` handling is supposed to do. If it is IDE/code editor specific, it does not work on VS code and I don't think we use such a pattern in the doc strings.

Reply:
@ialidzhikov 
Per offline discussion, I dropped the `()` part so the comments now use references it the form `[component.Deployer.Destroy]`. That should be standard godoc and in theory should work with any IDE which has full godoc support. WDYT?